x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_upgrade_modal/index.tsx

```diff
@@ -178,11 +178,10 @@ export const AgentUpgradeAgentModal: React.FunctionComponent<AgentUpgradeAgentMo
       value: option,
     }));
     if (options.length === 0) {
-      return [{ label: '', value: '' }];
+      return [{ label: kibanaVersion, value: kibanaVersion }];
```

Comment:
added a fallback to populate the combo list with kibana version if there is no versions available, this is to prevent a UI error if the combo doesn't have any versions and we open it.

<img width="1171" alt="image" src="https://github.com/elastic/kibana/assets/90178898/f07b1fea-28a8-4a63-95a0-e8b2268e6aac">

Reply:
I think this could be problematic in serverless where the kibana version will be a non existing agent version

Reply:
Good idea, added an input box instead if there are no versions:

<img width="705" alt="image" src="https://github.com/elastic/kibana/assets/90178898/f8da318b-ac6d-4420-8ac0-a47ba81934e9">